### PR TITLE
Mypy fixes

### PR DESCRIPTION
**Pull request by SuyogSoti**

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/core/azure-core/azure/core/tracing/decorator.py [hunk 1]
```diff
@@ -31,15 +31,23 @@
...
+
 
 def distributed_trace(func=None, name_of_span=None):
-    # type: (Callable[[Any], Any], str) -> Callable[[Any], Any]
+    # type: (Callable, str) -> Callable[[Any], Any]
     if func is None:
         return functools.partial(distributed_trace, name_of_span=name_of_span)
 
     @functools.wraps(func)
-    def wrapper_use_tracer(self, *args, **kwargs):
```
Comment (bryevdv) [inline sdk/core/azure-core/azure/core/tracing/decorator.py]: Oh, good. I noticed the other day and forgot to mention it. `*args, **kwarg`` is almost always the right signature for the inner wrapper

File: sdk/core/azure-core/azure/core/tracing/decorator.py [hunk 2]
```diff
@@ -50,18 +58,20 @@ def wrapper_use_tracer(self, *args, **kwargs):
         ans = None
         if parent_span is not None and orig_wrapped_span is None:
             common.set_span_contexts(parent_span)
-            name = name_of_span or self.__class__.__name__ + "." + func.__name__
+            name = (
+                name_of_span or getattr(func, "__qualname__", None) or common.get_function_and_class_name(func, *args)
```
Comment (bryevdv) [inline sdk/core/azure-core/azure/core/tracing/decorator.py]: why not put the getattr check in `get_function_and_class_name` and just call that only?

File: sdk/core/azure-core/azure/core/tracing/common.py [hunk 3]
```diff
@@ -36,11 +36,27 @@
...
+    is `self`. If there are no arguments then it only returns the function name.
+
+    :param func: the function passed in
+    :type func: `collections.abc.Callable`
+    :param args: List of arguments passed into the function
+    :type args: List[Any]
+    """
+    if args:
+        return "{}.{}".format(args[0].__class_.__name__, func.__name__) # pylint: disable=protected-access
+    return func.__name__
```
Comment (bryevdv) [inline sdk/core/azure-core/azure/core/tracing/common.py]: public API should get a test

Comment (SuyogSoti) [reply sdk/core/azure-core/azure/core/tracing/common.py]: done

File: sdk/core/azure-core/azure/core/tracing/decorator.py (hunk 2 above)
Comment (SuyogSoti) [reply sdk/core/azure-core/azure/core/tracing/decorator.py]: DONE

**Comment by SuyogSoti** [comment]
image

The first six errors are because of a bug in mypy: https://github.com/python/mypy/issues/2608 I made an internal issue to keep track of it #6586 

The last 6 errors are because the opencensus typings are not in the mypy typeshed libraries and opencensus does not install it when the customer installs opencensus.

**Pull request merged**